lib/streamlit/watcher/LocalSourcesWatcher.py

```diff
@@ -161,6 +161,14 @@ def _deregister_watcher(self, filepath):
         wm.watcher.close()
         del self._watched_modules[filepath]
 
+    def _file_should_be_watched(self, filepath):
+        file_is_new = filepath not in self._watched_modules
+        file_is_in_pythonpath = file_util.file_in_pythonpath(filepath)
+        file_is_local = file_util.file_is_in_folder_glob(
+            filepath, self._report.script_folder
+        )
+        return (file_is_local or file_is_in_pythonpath) and file_is_new
```

Comment:
I'd also add short-circuiting logic here _but_ that's just a performance nit that not everyone will probably agree with.

Reply:
Same as above... let me think about it...

Reply:
And here.